File: dotnet/PaymentService/Delivery.Platform.PaymentService/Events/Contracts/PaymentCompletedEvent.cs
using Delivery.Platform.Domain.Events;

namespace Delivery.Platform.PaymentService.Events.Contracts;

public sealed record PaymentCompletedEvent(
    Guid DeliveryId,
    Guid PaymentId,
    string ExternalTransactionId,
    decimal Amount,
    string Currency,
    Guid AggregateId,

    Guid EventId,
    DateTimeOffset OccurredAt,
    string Module = "payments",
    string EventType = PaymentEventNames.Completed) : IEvent
{

    public static PaymentCompletedEvent Create(
        Guid deliveryId,
        Guid paymentId,
        string externalTransactionId,
        decimal amount,
        string currency
    ) => new (deliveryId,
        paymentId,
        externalTransactionId,
        amount,
        currency,
        paymentId,
        EventId: Guid.NewGuid(),
        OccurredAt: DateTimeOffset.UtcNow);

}

File: libraries/common/src/main/kotlin/com/pl/platform/common/messaging/event/payments/PaymentCompletedEvent.kt
package com.pl.platform.common.messaging.event.payments

import java.math.BigDecimal
import java.util.UUID

data class PaymentCompletedEvent(
    override val paymentId: UUID,
    override val deliveryId: UUID,
    val externalTransactionId: String,
    val amount: BigDecimal,
    val currency: String,
)
    : PaymentEvent(
        paymentId = paymentId,
        deliveryId = deliveryId,
        eventType = PaymentEventType.Completed.value
    ){
}